pages/pipelines/command_step.md.erb

```diff
@@ -2,26 +2,113 @@
 
 ```yml
-- command: "command.sh"
+steps:
+  - command: "command.sh"
 ```
 
-The following step includes every configuration property for *command* steps:
+
+## Command Step Attributes
+
+_Optional attributes:_
+
+<table>
+  <tr>
+    <td><code>name</code></td>
+    <td>
+      The text and/or emoji that will be displayed in the pipeline visualisation in Buildkite.<br>
+      <em>Example:</em> <code>"\:hammer\: Tests"</code><br>
+    </td>
+  </tr>
+  <tr>
+    <td><code>branches</code></td>
+    <td>
+      The <a href="/docs/pipelines/branch-configuration#branch-pattern-examples">branch pattern</a> defining which branches will include this command step in their builds.<br>
+      <em>Example:</em> <code>"master stable/*"</code>
+    </td>
+  </tr>
+  <tr>
+    <td><code>env</code></td>
+    <td>
+      A list of <a href="/docs/builds/environment-variables">environment variables</a> for this step. <br>
+      <em>Example:</em> <code>FOO: "bar"</code>
+    </td>
+  </tr>
+  <tr>
+    <td><code>agents</code></td>
+    <td>
+      A list of <a href="/docs/agent/cli-meta-data">meta-data</a> key value pairs to <a href="docs/agent/cli-start#agent-targeting">target specific agents</a> for this step. <br>
+      <em>Example:</em> <code>npm: "true"</code>
+    </td>
+  </tr>
+  <tr>
+    <td><code>plugins</code></td>
+    <td>
+      A list of plugins that you require for this step. <br>
+      <em>Example:</em> <code>docker-compose</code><br>
+      <em>Example:</em> <code>harrietgl/myplugin</code>
+    </td>
+  </tr>
+  <tr>
+    <td><code>artifact_paths</code></td>
+    <td>
+      The path or paths where <a href="/docs/agent/cli-artifact">artifacts</a> from this step will be uploaded. <br>
+      <em>Example:</em> <code>"logs/**/*;coverage/**/*"</code>
+    </td>
+  </tr>
+  <tr>
+    <td><code>parallelism</code></td>
+    <td>
+      The number of jobs that will run this step <a href="/docs/builds/parallelizing-builds#job-parallelism">simultaneously</a>. <br>
+      <em>Example:</em> <code>1</code>
+    </td>
+  </tr>
+  <tr>
+    <td><code>concurrency</code></td>
+    <td>
+      The <a href="/docs/builds/controlling-concurrency#concurrency-limits">maximum number of jobs</a> that can be run inside this concurrency group at the same time as this job. <br>
+      <em>Example:</em> <code>1</code>
+    </td>
+  </tr>
+  <tr>
+    <td><code>concurrency_group</code></td>
+    <td>
+      A nickname for the concurrency value that you set with the <code>concurrency</code> attribute. This can also be the name of an existing <a href="/docs/builds/controlling-concurrency#concurrency-groups">concurrency group</a>. <br>
+      <em>Example:</em> <code>"serial_tests"</code>
+    </td>
+  </tr>
+  <tr>
+    <td><code>timeout_in_minutes</code></td>
+    <td>
+      The number of minutes this step should wait for an agent before timing out. When the step hits this timeout without finding an agent, the step will fail. <br>
```

Comment:
Ah, jobs do have `cancelling` and `canceled` states, but that's for when you hit the "Cancel" button in the web UI. And it causes the agent to terminate the running command on the agent machine.

I thought this was how timeouts worked on the agent machine too… hence why I wanted to mention it. Because people will probably want to know "What happens to the running job when the timeout is reached?"

I thought if the command times out, the agent will forcefully terminate the running command on the agent machine, the job will then be marked as `timed_out`, and the build will be `failed`.

But after looking at the code, my assumptions are actually wrong :( The agent never terminates the job on the machine when the timeout is reached. The web UI marks the job as `timed_out` and the agent goes along it's merry way, still running the job. I think this might be a bug.

I might need @sj26’s input on this one.